### PR TITLE
Integrate Python prototype

### PR DESCRIPTION
**Pull request by yunkee-lee**
This should work as expected but I have not tested yet. How do I restart python server?

### CODE REVIEW

**Comment by SmallFundamentals** [comment]
Not that familiar with django, but from what I gathered in a 10min search, looks like

`sudo touch /etc/apache2/conf-available/mds.conf`

http://stackoverflow.com/questions/17058928/how-to-reload-new-update-in-django-project-with-apache-mod-wsgi
I know we're using mod uwsgi because http://dev.sanamobile.org/users/manual_install/#

File: src/mds/api/v1/api.py [hunk 1]
```diff
@@ -574,3 +574,144 @@ def convert_binary(binary):
     result, message = FFmpeg().convert(binary, type, extension)
     logging.debug('FFmpeg: %s %s' % (result, message))
     return result
+
+def get_binary_checksum(procedure_guid, element_id, file_size):
+    """Returns checksums of the BinaryResource object associated with an encounter
+
+    If binary data is empty (no chuncks have been transferred), it fills data
```
Comment (SmallFundamentals) [inline src/mds/api/v1/api.py]: Should this be `doesn't exist` instead?

File: src/mds/api/v1/api.py [hunk 2]
```diff
@@ -574,3 +574,144 @@ def convert_binary(binary):
...
+    """
+    try:
+        sp = SavedProcedure.objects.get(guid=procedure_guid)
+    except SavedProcedure.DoesNotExist:
+        logging.error("Couldn't register binary %s for %s -- the saved "
+                      "procedure does not exist."
+                      % (element_id, procedure_guid))
+        return
+
+    binary, _ = BinaryResource.objects.get_or_create(
```
Comment (SmallFundamentals) [inline src/mds/api/v1/api.py]: should this be `binary, created` instead? otherwise `created` is not defined

File: src/mds/api/v1/api.py [hunk 3]
```diff
@@ -574,3 +574,144 @@ def convert_binary(binary):
...
+        trace = traceback.extract_tb(tb)
+        result = False
+        message = ('SavedProcedure -> %s, BinaryResource -> %s: %s %s'
+                   % (binary_guid, sp_guid, etype, value))
+
+        for tbm in trace:
+            logging.error(tbm)
+        raise
+
+    return result, message
```
Comment (SmallFundamentals) [inline src/mds/api/v1/api.py]: are we returning anything here if the try doesn't fail?

**Comment by SmallFundamentals** [comment]
Skimmed through what I can, looks fine to me. Didn't really look at the helpers and boilerplate code

File: src/mds/api/v1/api.py (hunk 1 above)
Comment (yunkee-lee) [reply src/mds/api/v1/api.py]: Saving procedure, which must be called before sending chunks, already creates an empty binary file with size 0.

File: src/mds/api/v1/api.py (hunk 2 above)
Comment (yunkee-lee) [reply src/mds/api/v1/api.py]: Oops

File: src/mds/api/v1/api.py (hunk 3 above)
Comment (yunkee-lee) [reply src/mds/api/v1/api.py]: Good call. Another bad example of following the existing API.

**Pull request closed without merging**